Sources/SwiftSyntax/TokenKind.swift.gyb

```diff
@@ -31,6 +31,29 @@ public enum TokenKind {
   case ${kind}
 % end
 
+  /// Initializes a keyword token kind from its string representation. If the
+  /// given string is not a keyword, this function returns `nil`.
+  public init?(keyword: String) {
+    switch keyword.utf8.count {
+% from itertools import groupby
+% for length, items in groupby(sorted(list(filter(lambda x: x.is_keyword, SYNTAX_TOKENS)), key=lambda x: len(x.text)), key=lambda x: len(x.text)):
```

Comment:
If there's a better way to express this, I'm all ears.

Reply:
The idea is to take all the keywords, sort them into buckets by the number of characters they have, then take `(length, bucket)` pairs and render switch statements out of each bucket after testing the length.

Reply:
I assume there is there a measurable performance benefit from do the switch on the length first? If not, we could just keep it simple and switch on the string contents themselves.

Reply:
There is a significant benefit in my testing. But I agree let's keep things simple.